File: C++/2-Medium/00254._因子的组合/00254._因子的组合.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> getFactors(int n) {
        vector<vector<int>> result;
        vector<int> combination;
        backtrack(n, combination, result, 2);
        return result;
    }

private:
    void backtrack(int n, vector<int>& combination, vector<vector<int>>& result, int start) {
        if (n == 1) {
            if (combination.size() > 1) { // 确保至少有两个因子
                result.push_back(combination);
            }
            return;
        }
        
        for (int i = start; i <= n; ++i) {
            if (n % i == 0) { // 如果i是n的一个因子
                combination.push_back(i); // 添加因子到当前组合
                backtrack(n / i, combination, result, i); // 继续搜索剩余的因子
                combination.pop_back(); // 回溯，移除最后一个因子，尝试下一个可能的因子
            }
        }
    }
};

int main() {
    Solution solution;
    vector<vector<int>> result = solution.getFactors(16);
    for (const auto& factors : result) {
        for (int factor : factors) {
            cout << factor << " ";
        }
        cout << endl;
    }
    return 0;
}

File: Python/2-Medium/00254._因子的组合/00254._因子的组合.py
from typing import List

def get_factors(n: int) -> List[List[int]]:
    """
    寻找所有独特的因子组合，其乘积等于给定的 n。

    Args:
    n: 一个正整数。

    Returns:
    一个列表，包含所有独特的因子组合。
    
    示例:
    >>> get_factors(8)
    [[2, 2, 2], [2, 4]]
    """

    def backtrack(start: int, path: List[int], target: int) -> None:
        """
        递归回溯函数，用于找到所有因子组合。

        Args:
        start: 开始的因子。
        path: 当前因子组合的路径。
        target: 目标乘积值。
        """
        # 终止条件：如果目标乘积降到1，说明找到了一组有效组合
        if target == 1:
            # 需要至少两个因子组合，因此长度需大于1
            if len(path) > 1:
                result.append(path[:])
            return
        
        for i in range(start, target + 1):
            # 如果当前因子能整除目标值，尝试该因子
            if target % i == 0:
                # 添加当前因子到路径中
                path.append(i)
                # 递归调用，目标值除以当前因子
                backtrack(i, path, target // i)
                # 回溯，移除路径中的当前因子
                path.pop()

    result = []  # 存储所有可能的因子组合
    backtrack(2, [], n)  # 从因子2开始
    return result

# 测试代码
print(get_factors(8))  # [[2, 2, 2], [2, 4]]